apps/web/pages/api/auth/passkeys/register/finalize.ts

```diff
@@ -0,0 +1,41 @@
+import { tenant } from "@teamhanko/passkeys-next-auth-provider";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { ErrorCode } from "@calcom/features/auth/lib/ErrorCode";
+import { getServerSession } from "@calcom/features/auth/lib/getServerSession";
+import prisma from "@calcom/prisma";
+
+const tenantId = process.env.NEXT_PUBLIC_HANKO_PASSKEYS_TENANT_ID ?? "";
+const apiKey = process.env.HANKO_PASSKEYS_API_KEY ?? "";
+
+const passkeyApi = tenant({ tenantId, apiKey });
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (!tenantId || !apiKey) {
+    return res.status(503).json({ message: "Passkey API not configured" });
```

Comment:
Is 503 the best status code to use here?

Reply:
Other routes use 403: https://github.com/calcom/cal.com/blob/main/apps/web/pages/api/auth/signup.ts#L23

However, [MDN says](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503)

> [...] 503 Service Unavailable server error response code indicates that the server is not ready to handle the request.
>
> Common causes are a server that is down for maintenance or that is overloaded. This response should be used for temporary conditions [...]

While in my mind 403 has always been more for insufficient permissions etc

What do you think?

Reply:
I don't think a 403 is correct here. 501 is used for not implemented.

Reply:
I'd prefer 403 Forbidden, 503 implies a server issue but most of the time this'll be a client issue (don't access the route)

Reply:
Don't think a 403 is correct here though because that leads the client to believe they are lacking permissions when in reality the functionality isn't supported at all. Would consider 501 or 405 instead.

Reply:
See https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501

Reply:
If everyone is fine with it, will go with 501 - seems like the most appropriate IMO